frontend/src/pages/instance/Snapshots/index.vue

```diff
@@ -259,13 +256,15 @@ export default {
             this.$emit('instance-updated')
         },
         async downloadSnapshotPackage (snapshot) {
-            const ss = await SnapshotApi.getSnapshot(this.instance.id, snapshot.id)
+            const ss = await SnapshotsApi.getFullSnapshot(snapshot.id)
+            const owner = snapshot.device || snapshot.project
+            const ownerType = snapshot.device ? 'device' : 'instance'
```

Comment:
This transformation/mapping seems similar to the ones in the previous components. Would it be possible to extract the entire download method into a composable/mixin?

Reply:
resolved in [30d8d50](https://github.com/FlowFuse/flowfuse/pull/3885/commits/30d8d50572972a1d4486fd848882f640385db857)